webpack/assets/javascripts/react_app/components/AutoCompleteSearch/AutoCompleteSearchBox.js

```diff
@@ -0,0 +1,77 @@
+import React from 'react';
+import debounce from 'lodash/debounce';
+import { Icon, TypeAheadSelect } from 'patternfly-react';
+import './auto-complete.scss';
+import { STATUS } from '../../constants';
+import helpers from '../../common/helpers';
+import { renderMenu, filterResults } from './helpers';
+
+class AutoCompleteSearchBox extends React.Component {
+  constructor(props) {
+    super(props);
+    this.handleInputChange = debounce(this.handleInputChange, 250);
```

Comment:
can you use `DebounceInput` from `'react-debounce-input' 
 you can see its usage at `SearchInput` componen

Reply:
I don't think this will work in our case because the `react-bootstrap-typeahead` 
events are raised by inner-state updates nor window events.. 
Another problem is that we can't wrap or replace the inner input field with `DebounceInput`.